plugins/autowarmth.koplugin/main.lua

```diff
@@ -0,0 +1,777 @@
+    local function store_times(touchmenu_instance, new_time, num)
+        self.scheduler_times[num] = new_time
+        if num == 1 then
+            if new_time then
+                self.scheduler_times[midnight_index]
+                    = new_time + 24 -- next day
+            else
+                self.scheduler_times[midnight_index] = nil
+            end
+        end
+        G_reader_settings:saveSetting("autowarmth_scheduler_times",
+            self.scheduler_times)
+        self:scheduleMidnightUpdate()
+        if touchmenu_instance then touchmenu_instance:updateItems() end
+    end
+    -- mode == nil ... show alway
+    --      == true ... easy mode
+    --      == false ... expert mode
+    local function getScheduleMenuEntry(text, num, mode)
+        return {
+            mode = mode,
+            text_func = function()
+                return T(_"%1: %2", text,
+                    self:hoursToClock(self.scheduler_times[num], true))
+            end,
+            checked_func = function()
+                return self.scheduler_times[num] ~= nil
+            end,
+            callback = function(touchmenu_instance)
+                local hh = 12
+                local mm = 0
+                if self.scheduler_times[num] then
+                    hh = math.floor(self.scheduler_times[num])
+                    mm = math.floor(frac(self.scheduler_times[num]) * 60 + 0.5)
+                end
+                UIManager:show(DoubleSpinWidget:new{
+                    title_text = _("Set time"),
+                    left_text = _("HH"),
+                    left_value = hh,
+                    left_default = 0,
+                    left_min = 0,
+                    left_max = 23,
+                    left_step = 1,
+                    left_hold_step = 3,
+                    left_wrap = true,
+                    right_text = _("MM"),
+                    right_value = mm,
+                    right_default = 0,
+                    right_min = 0,
+                    right_max = 59,
+                    right_step = 1,
+                    right_hold_step = 5,
+                    right_wrap = true,
+                    callback = function(left, right)
+                        local new_time = left + right / 60
+                        local function get_valid_time(n, dir)
+                            for i = n+dir, dir > 0 and midnight_index or 1, dir do
+                                if self.scheduler_times[i] then
+                                    return self.scheduler_times[i]
+                                end
+                            end
+                            return dir > 0 and 0 or 26
+                        end
+                        if num > 1 and new_time < get_valid_time(num, -1) then
+                            UIManager:show(ConfirmBox:new{
+                                text =  _("This time conflicts with the previous time.\nShall the previous time be adjusted?"),
```

Comment:
```suggestion
                                text =  _("This time conflicts with the previous time.\nAdjust the previous time?"),
```

Instead of phrasing along the lines of "conflicts with" it'd probably be clearer to say something along these lines?

"The start/end time should be before/after the start/end time. Adjust the start time."

Reply:
I would want to set it the way 
``` 
text =  _("This time is before the previous time.\nAdjust the previous time?"),
```
as this is what a `yes` would do. (We could talk about "Adjust the previous time(s)" but that would make it even more complicated. The user will find out, what this does :) )